File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-fortran-skills/tsvc_2_vtvtv__submitted.f90
subroutine tsvc_2_vtvtv_fp64(a, b, c, len_1d, workspace, workspace_size) bind(C)
  use iso_c_binding
  implicit none
  real(c_double), intent(inout) :: a(*)
  real(c_double), intent(in) :: b(*), c(*)
  integer(c_int64_t), value, intent(in) :: len_1d
  integer(c_int8_t), intent(inout) :: workspace(*)
  integer(c_int64_t), value, intent(in) :: workspace_size
  integer(c_int64_t) :: i
  !$omp parallel do simd
  do i = 1, len_1d
    a(i) = a(i) * b(i) * c(i)
  end do
  !$omp end parallel do simd
end subroutine tsvc_2_vtvtv_fp64

File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-c-skills/tsvc_2_vtvtv__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_vtvtv_fp64(double * restrict a, const double * restrict b, const double * restrict c, int64_t LEN_1D, uint8_t * restrict workspace, int64_t workspace_bytes) {
    (void)workspace; // unused
    (void)workspace_bytes;
    #pragma omp parallel for simd schedule(static)
    for (int64_t i = 0; i < LEN_1D; ++i) {
        a[i] = a[i] * b[i] * c[i];
    }
}

